pkg/bytequantity/bytequantity.go

```diff
@@ -0,0 +1,152 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package bytequantity
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	/// Examples:          1mb, 1 gb, 1.0tb, 1mib, 2g, 2.001 t
+	byteQuantityRegex = `^([0-9]+\.?[0-9]{0,3})[ ]?(mi?b?|gi?b?|ti?b?)?$`
+	mib               = "MiB"
+	gib               = "GiB"
+	tib               = "TiB"
+	gbConvert         = 1 << 10
+	tbConvert         = gbConvert << 10
+	maxGiB            = math.MaxUint64 / gbConvert
+	maxTiB            = math.MaxUint64 / tbConvert
+)
+
+// ByteQuantity is a data type representing a byte quantity
+type ByteQuantity struct {
+	Quantity uint64
+}
+
+// ParseToByteQuantity parses a string representation of a byte quantity to a ByteQuantity type
+func ParseToByteQuantity(byteQuantityStr string) (ByteQuantity, error) {
+	bqRegexp := regexp.MustCompile(byteQuantityRegex)
+	matches := bqRegexp.FindStringSubmatch(strings.ToLower(byteQuantityStr))
+	if len(matches) < 2 {
+		return ByteQuantity{}, fmt.Errorf("%s is not a valid byte quantity", byteQuantityStr)
+	}
+
+	quantityStr := matches[1]
+	unit := gib
```

Comment:
super nit: code reads as if mib is the default

Reply:
I'm not sure what you mean here? Do you just mean because it's storing it in MiB?  It's stored in MiB since that's the lowest unit supported so we can store it in an integer without floating point. The default unit for parsing is still GiB if no unit is specified. Maybe I should add that in the comment for the method :)